### PR TITLE
Supporting incidents in watchguard epdr

### PR DESCRIPTION
**Pull request by LenaigKaliou**
Related to [this issue](https://github.com/SekoiaLab/integration/issues/989)

## Summary by Sourcery

Implement support for incidents in the WatchGuard EPDR integration by extending the HTTP client and connector, integrating incident processing into the main loop, and adding comprehensive tests and changelog updates.

New Features:
- Add fetch_incidents method to WatchGuardClient for retrieving incidents from the API.
- Introduce get_watchguard_incidents and last_incident_date in WatchGuardEpdrConnector to fetch, cache, and push incidents.
- Include incident counts in the EPDR connector run loop alongside events.

Enhancements:
- Add caching logic for last incident timestamp with a default 24-hour fallback.
- Extend epdr_connector test fixture to accept a temporary storage path.

Documentation:
- Update CHANGELOG.md to document incident support in version 1.1.4.

Tests:
- Add connector tests for incident fetching, empty responses, and caching behaviors.
- Add HTTP client tests for successful incident fetch and auth-retry scenarios.

### CODE REVIEW

File: WatchGuard/watchguard/epdr_connector.py [hunk 1]
```diff
@@ -143,7 +216,9 @@ def run(self) -> None:  # pragma: no cover
                             processing_start - previous_processing_end
                         )
 
-                    events_count = loop.run_until_complete(self.get_watchguard_events())
+                    events_count = loop.run_until_complete(self.get_watchguard_events()) + loop.run_until_complete(
+                        self.get_watchguard_incidents()
+                    )
```
[the remark this answers is not in the document]
Comment (TOUFIKIzakarya) [reply WatchGuard/watchguard/epdr_connector.py]: I think this comment is relevant

**Comment by squioc** [comment]
Hello,

I close this PR. The incident APi belongs to WatchGuard ThreatSync, that is a different product than WatchGuard EDPR.
